applications/serial_lte_modem/doc/slm_description.rst

```diff
@@ -0,0 +1,818 @@
+         PONG: Test UDP by hostname
+
+   #. Disconnect from the server.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPCLI=0**
+         #XUDPCLI: disconnected
+         OK
+
+
+TLS client
+~~~~~~~~~~
+
+Before completing this test, you must update the CA certificate, the client certificate, and the private key to be used for the TLS connection in the modem.
+The credentials must use the security tag 16842755.
+
+To store the credentials in the modem, use LTE Link Monitor.
+See `Managing credentials`_ in the LTE Link Monitor User Guide for instructions.
+
+You must register the corresponding credentials on the server side.
+
+1. Establish and test a TLS connection:
+
+   a. List the credentials that are stored in the modem with security tag 16842755:
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT%CMNG=1,16842755**
+         %CMNG: 16842755,0,"0000000000000000000000000000000000000000000000000000000000000000"
+         %CMNG: 16842755,1,"0101010101010101010101010101010101010101010101010101010101010101"
+         %CMNG: 16842755,2,"0202020202020202020202020202020202020202020202020202020202020202"
+         OK
+
+   #. Open a TCP/TLS socket that uses the security tag 16842755 and connect to a TLS server on a specified port.
+      Replace *example.com* with the host name or IPv4 address of a TLS server and *1234* with the corresponding port.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XSOCKET=1,1,0,16842755**
+         #XSOCKET: 2, 1, 0, 258
+         OK
+
+         **AT#XCONNECT="**\ *example.com*\ **",**\ *1234*
+         #XCONNECT: 1
+         OK
+
+   #. Send plain text data to the TLS server and retrieve the returned data.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XSEND=1,"Test TLS client"**
+         #XSEND: 15
+         OK
+
+         **AT#XRECV**
+         PONG: b'Test TLS client'
+         #XRECV: 1, 24
+         OK
+
+   #. Close the socket.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XSOCKET=0**
+         #XSOCKET: 0, closed
+         OK
+
+#. Test a TLS client with TCP proxy service (note that these commands are available only if :option:`CONFIG_SLM_TCP_PROXY` is defined):
+
+   a. Create a TCP/TLS client and connect to a server.
+      Replace *example.com* with the host name or IPv4 address of a TLS server and *1234* with the corresponding port.
+      Then read information about the connection.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XTCPCLI=1,"**\ *example.com*\ **",**\ *1234*
+         #XTCPCLI: 2 connected
+         OK
+
+         **AT#XTCPCLI?**
+         #XTCPCLI: 1
+         OK
+
+   #. Send plain text data to the TLS server and retrieve the returned data.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XTCPSEND=1,"Test TLS client"**
+         #XTCPSEND: 15
+         OK
+         #XTCPDATA: 1, 24
```

Comment:
Is this returned after the "OK"?

Reply:
Again, this is an unsolicited event of data receiving. We tested with the Nordic server, which is a ping-pong server. So after Send, Receive happens immediately